src/clients/python/src/tigerbeetle/client.py

```diff
@@ -280,6 +281,32 @@ async def _submit(self, operation: bindings.Operation, operations: Any,
 
         return inflight_packet.response
 
+    async def close(self) -> None:
+        tb_assert(self._client is not None)
+        bindings.tb_client_deinit(ctypes.byref(self._client))
+
+        # tb_client_deinit internally clears any inflight requests, and calls their callbacks, so
+        # the client needs to stick around until that's done.
+        #
+        # This isn't a problem for ClientSync, but ClientAsync invokes the callbacks using
+        # call_soon_threadsafe(), so wait for the event to fire on all pending packets.
+        all_events = []
+        for packet in self._inflight_packets.values():
+            if not isinstance(packet.on_completion_context, CompletionContextAsync):
+                raise AssertionError()
+            all_events.append(packet.on_completion_context.event.wait())
```

Comment:
Does `all_events.append(packet.on_completion_context.event.wait())` append a **future** that will be awaited by `await asyncio.gather(*all_events)`, or does it **block** on each iteration (and in this case, the `gather()` call is unnecessary)?

From the `event.wait()` documentation:

> Block until the internal flag is true.

Should we use `asyncio.create_task` instead?

```python
task = asyncio.create_task(packet.on_completion_context.event.wait())
all_events.append(task)
```

Reply:
Although... I actually don't see any benefit to having the `.gather()` as opposed to just `await`ing in the loop.